### PR TITLE
Move VIN generators to VehicleCommon.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: common/src/main/scala/org/genivi/sota/datatype/Vehicle.scala [hunk 1]
```diff
@@ -22,4 +23,31 @@ trait VehicleCommon {
     override def compare(v1: Vin, v2: Vin): Int = v1.get compare v2.get
   }
 
+  val genVin: Gen[Vin] =
```
Comment (jerrytrieu) [inline common/src/main/scala/org/genivi/sota/datatype/Vehicle.scala]: Can all these gens be in their own separate trait?

File: core/src/main/scala/org/genivi/sota/core/data/Vehicle.scala [hunk 2]
```diff
@@ -16,4 +17,13 @@ object Vehicle extends VehicleCommon {
 
   val MissingVehicle = new ErrorCode("missing_vehicle")
 
+  val genVehicle: Gen[Vehicle] =
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/data/Vehicle.scala]: This still needs to be duplicated in resolver?

File: common/src/main/scala/org/genivi/sota/datatype/Vehicle.scala (hunk 1 above)
Comment (stevana) [inline common/src/main/scala/org/genivi/sota/datatype/Vehicle.scala]: That would be nice; my naive attempts fail to compile... @koshelev?

File: core/src/main/scala/org/genivi/sota/core/data/Vehicle.scala (hunk 2 above)
Comment (stevana) [inline core/src/main/scala/org/genivi/sota/core/data/Vehicle.scala]: Not as it stands, but in the future when the notion of `Vehicle` will be different between core and resolver there will have to be some duplication -- so I set things up anticipating that.

**Pull request merged**